scene/resources/particles_material.cpp

```diff
@@ -261,6 +269,195 @@ void ParticlesMaterial::_update_shader() {
 	code += "}\n";
 	code += "\n";
 
+	//functions to create simplex noise, curl noise in 3D and 4D
+	code += "\n\n";
+	code += "vec4 permute(vec4 x) {\n";
+	code += "	return mod(((x * 34.0) + 1.0) * x, 289.0);\n";
+	code += "}\n";
+	code += "\n";
+
+	code += "\n\n";
+	code += "vec4 taylor_inv_sqrt(vec4 r) {\n";
+	code += "return 1.79284291400159 - 0.85373472095314 * r;\n";
+	code += "}\n";
+	code += "\n";
```

Comment:
You missed a tab here. :)

Also, I don't think there's a need to have four newlines in a row (two from the end of this one, and two from the beginning of the next one).